RUFAS/biophysical/manure/separators/separator.py

```diff
@@ -0,0 +1,133 @@
+from dataclasses import dataclass
+from RUFAS.biophysical.manure.processor import Processor
+from RUFAS.current_day_conditions import CurrentDayConditions
+from RUFAS.data_structures.animal_to_manure_connection import ManureStream
+from RUFAS.time import Time
+
+
+@dataclass
+class SeparatorConfig:
+    """
+    Base configuration for a manure separator.
+
+    Attributes
+    ----------
+    water_efficiency : float
+        The efficiency of the separator in removing water from the manure.
+    ammoniacal_nitrogen_efficiency : float
+        The efficiency of the separator in removing ammoniacal nitrogen from the manure.
+    nitrogen_efficiency : float
+        The efficiency of the separator in removing nitrogen from the manure.
+    phosphorus_efficiency : float
+        The efficiency of the separator in removing phosphorus from the manure.
+    potassium_efficiency : float
+        The efficiency of the separator in removing potassium from the manure.
+    ash_efficiency : float
+        The efficiency of the separator in removing ash from the manure.
+    non_degradable_volatile_solids_efficiency : float
+        The efficiency of the separator in removing non-degradable volatile solids from the manure.
+    degradable_volatile_solids_efficiency : float
+        The efficiency of the separator in removing degradable volatile solids from the manure.
+    total_solids_efficiency : float
+        The efficiency of the separator in removing total solids from the manure.
+    """
+
+    water_efficiency: float
+    ammoniacal_nitrogen_efficiency: float
+    nitrogen_efficiency: float
+    phosphorus_efficiency: float
+    potassium_efficiency: float
+    ash_efficiency: float
+    non_degradable_volatile_solids_efficiency: float
+    degradable_volatile_solids_efficiency: float
+    total_solids_efficiency: float
+
+
+class Separator(Processor):
+    """
+    A manure processor that separates solids from liquids.
```

Comment:
```suggestion
    A manure processor that separates a portion of solids from manure.
```
(Sorry, this is me being pedantic, but what else are SMEs for 😅)

Reply:
your pedantry is a gift to the project :).

Reply:
> your pedantry is a gift to the project :).

I second this